axelrod/tournament.py

```diff
@@ -48,19 +48,23 @@ def __init__(self, players, match_generator=RoundRobinMatches,
             self.game = game
         self.players = players
         self.repetitions = repetitions
-        if deterministic_cache is None:
-            self.deterministic_cache = DeterministicCache()
-        else:
-            self.deterministic_cache = deterministic_cache
-        self.match_generator = match_generator(
-            players, turns, self.game, self.deterministic_cache)
+        self.match_generator = match_generator(players, turns, self.game,
+                                               self.repetitions)
         self._with_morality = with_morality
         self._parallel_repetitions = repetitions
         self._processes = processes
         self._logger = logging.getLogger(__name__)
-        self.interactions = []
+        self.interactions = defaultdict(list)
 
-    def play(self, filename=None):
+        if not filename:
+            self.outputfile = NamedTemporaryFile(mode='w')
+            filename = self.outputfile.name
+        else:
+            self.outputfile = open(filename, 'w')
+        self.writer = csv.writer(self.outputfile)
+        self.filename = filename
+
+    def play(self):
```

Comment:
I would have a warning or perhaps if `filename=None` then we automatically set `build_results` to True with perhaps a warning as well?